### PR TITLE
Fix styling issues found in example-app-base cleanup

### PR DESCRIPTION
**Pull request by ecraig12345**
#### Pull request checklist

- [x] Include a change request file using `$ npm run change`

#### Description of changes

Fixing some styling-related issues found during example-app-base cleanup:
- remove CSS global utility classes defined by ExampleCard
- make FeedbackList not depend on some random class names from the list ghosting example
- make various places use built-in constants and functions for handling high contrast, rather than rolling their own

#### Focus areas to test

- [x] Test all modified components/examples in high contrast mode

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8467)

### CODE REVIEW

**Comment by ecraig12345** [comment]
In case anyone is wondering, here are the definitions for [HighContrastSelector](https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/styling/src/styles/CommonStyles.ts) and [getFocusStyle](https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/styling/src/styles/getFocusStyle.ts).

File: packages/example-app-base/src/components/FeedbackList/FeedbackList.styles.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { getTheme, mergeStyleSets, DefaultFontStyles, getFocusStyle } from 'office-ui-fabric-react/lib/Styling';
+
+const theme = getTheme();
+export const classNames = mergeStyleSets({
```
Comment (ecraig12345) [inline packages/example-app-base/src/components/FeedbackList/FeedbackList.styles.ts]: So far I'm converting *only* the former `ms-ListGhostingExample-*` classes to css-in-js. The rest of the conversions will come later.

File: packages/charting/src/components/Legends/Legends.styles.ts [hunk 2]
```diff
@@ -20,7 +21,7 @@ export const getStyles = (props: ILegendStyleProps): ILegendsStyles => {
     },
     rect: {
       selectors: {
-        '@media screen and (-ms-high-contrast: active)': {
+        [HighContrastSelector]: {
```
Comment (ecraig12345) [inline packages/charting/src/components/Legends/Legends.styles.ts]: [HighContrastSelector definition](https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/styling/src/styles/CommonStyles.ts) for reference (it's identical to this)

File: packages/office-ui-fabric-react/src/components/Coachmark/PositioningContainer/PositioningContainer.styles.ts [hunk 3]
```diff
@@ -21,41 +20,22 @@ export interface IPositioningContainerNames {
...
+          position: 'absolute',
+          boxSizing: 'border-box',
+          border: '1px solid ${}',
+          selectors: {
+            [HighContrastSelector]: {
+              border: '1px solid WindowText'
+            }
+          }
+        },
+        focusClear()
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/Coachmark/PositioningContainer/PositioningContainer.styles.ts]: [focusClear definition](https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/styling/src/styles/getFocusStyle.ts) for reference (it's functionally identical to what the custom method was doing)

File: packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Basic.Example.tsx [hunk 4]
```diff
@@ -7,32 +7,41 @@ import { Link } from 'office-ui-fabric-react/lib/Link';
...
-          <PivotItem headerText="Pivot 1" keytipProps={keytipMap.Pivot1Keytip} style={{ width: 500, paddingTop: 20 }}>
-            <SpinButton className="u-marginBottom" label={'Spin Button'} keytipProps={keytipMap.SpinButtonKeytip} />
-            <Toggle className="u-marginBottom" keytipProps={keytipMap.ToggleKeytip} onText={'Yes'} offText={'No'} />
-            <p>
-              Go to{' '}
-              <Link keytipProps={keytipMap.LinkKeytip} href={'http://www.bing.com'} target="_blank">
-                Bing
-              </Link>
-            </p>
+          <PivotItem headerText="Pivot 1" keytipProps={keytipMap.Pivot1Keytip} style={pivotItemStyle}>
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Basic.Example.tsx]: This usage of `u-marginBottom` was better handled with a Stack

File: packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Button.Example.tsx [hunk 5]
```diff
@@ -2,11 +2,15 @@ import * as React from 'react';
...
 import { keytipMap } from './KeytipSetup';
 import { Toggle } from 'office-ui-fabric-react/lib/Toggle';
+import { Stack } from 'office-ui-fabric-react/lib/Stack';
+import { mergeStyles } from 'office-ui-fabric-react/lib/Styling';
 
 export interface IKeytipsButtonExampleState {
   btnDisabled: boolean;
 }
 
+const marginBottom = mergeStyles({ marginBottom: 28 });
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Button.Example.tsx]: Recommend enabling "Hide whitespace changes" to view this file...the actual changes are:
1. use a Stack rather than `u-marginRight` for horizontal spacing
2. use this button style rather than `u-marginBottom`

File: packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Button.Example.tsx [hunk 6]
```diff
@@ -21,78 +25,81 @@ export class KeytipsButtonExample extends React.Component<{}, IKeytipsButtonExam
...
+                {
+                  key: 'buttonMenuItem2',
+                  text: 'Menu Item 2',
+                  keytipProps: keytipMap.ButtonMenuItem2
+                }
+              ]
+            }}
+          />
+          <DefaultButton
+            styles={{ splitButtonContainer: { height: 32 } }}
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Button.Example.tsx]: And the other actual change (3) is giving the split button a height--for some reason after changing the other styles, it was filling the height of the container rather than just being the same height as its main button.

File: packages/office-ui-fabric-react/src/components/List/examples/List.Ghosting.Example.tsx [hunk 7]
```diff
@@ -2,25 +2,70 @@ import * as React from 'react';
...
-  constructor(props: IListGhostingExampleProps) {
-    super(props);
+const theme = getTheme();
+const classNames = mergeStyleSets({
+  container: {
+    overflow: 'auto',
+    maxHeight: 500
+  },
+  itemCell: [
+    getFocusStyle(theme, -1),
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/List/examples/List.Ghosting.Example.tsx]: [getFocusStyle](https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/styling/src/styles/getFocusStyle.ts) does almost exactly the same thing as the `focus-border` mixin, though with slightly different parameter defaults.

File: packages/example-app-base/src/components/FeedbackList/FeedbackList.styles.ts (hunk 1 above)
Comment (JasonGore) [reply packages/example-app-base/src/components/FeedbackList/FeedbackList.styles.ts]: Hmm, another option would be to replace the divs with Stacks and just use the `styles` prop...

File: packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Basic.Example.tsx [hunk 8]
```diff
@@ -7,32 +7,41 @@ import { Link } from 'office-ui-fabric-react/lib/Link';
...
+            <Stack gap={20}>
```
Comment (JasonGore) [inline packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Basic.Example.tsx]: Just an FYI mainly, #8247 adds `childrenGap` as added as a `token` replacing the `gap` prop. This means `gap={20}` sprinkled through this file could just be a centralized `token` object similar to `styles` objects.

File: packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Button.Example.tsx (hunk 5 above)
Comment (JasonGore) [reply packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Button.Example.tsx]: This seems to be used for `styles` props. Is `mergeStyles` needed? I feel like this instead could be:

```tsx
const marginBottomStyle = { root: { marginBottom: 28 } };
```

and just passed directly as the `styles` prop.

File: packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Basic.Example.tsx [hunk 9]
```diff
@@ -7,32 +7,41 @@ import { Link } from 'office-ui-fabric-react/lib/Link';
...
+              <SpinButton label="Spin Button" keytipProps={keytipMap.SpinButtonKeytip} styles={{ root: { maxWidth: 200 } }} />
```
Comment (khmakoto) [inline packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Basic.Example.tsx]: Why did you add the `styles={{ root: { maxWidth: 200 } }}` definition here?

File: packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Button.Example.tsx [hunk 10]
```diff
@@ -21,78 +25,81 @@ export class KeytipsButtonExample extends React.Component<{}, IKeytipsButtonExam
...
-                text: 'Split Button Menu Item 2',
-                keytipProps: keytipMap.SplitButtonMenuItem2
-              }
-            ]
-          }}
-        />
-        <DefaultButton text="I do not have a keytip" />
+        <Stack horizontal gap={20}>
+          <DefaultButton
+            className="u-marginRight"
```
Comment (khmakoto) [inline packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Button.Example.tsx]: We probably need to also take this out?

File: packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Basic.Example.tsx (hunk 8 above)
Comment (ecraig12345) [reply packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Basic.Example.tsx]: Good to know, though I'd like to go ahead and check this in rather than waiting on another PR (since I have multiple other changes waiting on this one)

**Comment by ecraig12345** [comment]
@JasonGore re: https://github.com/OfficeDev/office-ui-fabric-react/pull/8467#discussion_r269295523, not sure why I used mergeStyles here...I'll remove it.

File: packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Basic.Example.tsx (hunk 9 above)
Comment (ecraig12345) [reply packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Basic.Example.tsx]: Because a 400px wide SpinButton is ugly.
image

File: packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Button.Example.tsx (hunk 10 above)
Comment (ecraig12345) [reply packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Button.Example.tsx]: Good catch, thanks!

File: packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Button.Example.tsx (hunk 5 above)
Comment (ecraig12345) [reply packages/office-ui-fabric-react/src/components/Keytip/examples/Keytips.Button.Example.tsx]: Good point. Fixed that.

File: packages/example-app-base/src/components/FeedbackList/FeedbackList.styles.ts (hunk 1 above)
Comment (ecraig12345) [reply packages/example-app-base/src/components/FeedbackList/FeedbackList.styles.ts]: Not quite sure offhand what that would look like? I'd prefer to leave this as-is for this PR and possibly come back to it later.

**Comment by khmakoto** [comment]
@ecraig12345 it appears you need to update the API of the `merge-styles` package.

**Comment by ecraig12345** [comment]
@khmakoto That's actually due to (essentially) a non-obvious merge conflict from Natalie's api-extractor PR. It's being fixed in #8477.

**Comment by ecraig12345** [comment]
Updated snapshots to fix build failure.
